examples/vm-joined-to-active-directory/README.md

```diff
@@ -0,0 +1,87 @@
+# Example: Joining a Windows Machine to an Active Directory Domain
+
+This example creates an Active Directory Domain, a Windows Client; to demonstrate how to bind a Windows Client to an Active Directory Domain using a Virtual Machine Extension in Terraform (using [the `azurerm_virtual_machine_extension` resource](https://www.terraform.io/docs/providers/azurerm/r/virtual_machine_extension.html)).
+
+This example is built around [the Virtual Machine Extension](https://www.terraform.io/docs/providers/azurerm/r/virtual_machine_extension.html) found in the `windows-client` module (documented below) - which demonstrates binding a Windows Virtual Machine to an Active Directory Domain. For the purposes of this example we create an Active Directory Domain, since it's easier to demonstrate - however you can achieve the same thing with an existing Active Directory Domain.
+
+## Notes
+
+- This is intended as an example of binding machines to an Active Directory Domain, and **we'd not recommend using it in production as-is** as the configuration has been simplified for example purposes, e.g.:
+  - The Active Directory Forest has a single node, for demonstration purposes
+  - There's no security rules configured on the network, so everything's open internally etc.
+- The numbering on the files within the modules below have no effect on which order the resources are created in - it's purely to make the examples easier to understand.
+
+## Running this Example
+
+Initialize the modules (and download the Azure Provider) by running `terraform init`:
+
+```bash
+$ terraform init
+```
+
+In order to run this example you'll need some kind of credentials configured - either a Service Principal or to be logged into the Azure CLI. You can find out more about this on [the Azure Provider overview page](https://www.terraform.io/docs/providers/azurerm/index.html)
+
+Once you've initialized the Provider - you can run the sample by running:
+
+```bash
+$ terraform apply
+```
+
+This will take around 20m to provision - once completed you should see the Public IP Address of the Windows Client machine (which is bound to the Active Directory Domain):
+
+```
+```
+
+## Variables
+
+ * `prefix` - The prefix used for all resources in this example. Needs to be a short (6 characters) alphanumeric string. Example: `myprefix`.
```

Comment:
nitpick: you say it needs to be 6 characters, and the example is 8.

Reply:
🤦‍♂️ good spot